pkg/proctree/fileinfo.go

```diff
@@ -0,0 +1,190 @@
+package proctree
+
+import (
+	"sync"
+	"time"
+)
+
+//
+// File Info
+//
+
+// FileInfo represents a file.
+type FileInfo struct {
```

Comment:
Yep, the correct way of doing it would be to create a type with public fields and copy the information from the internal fields and give to caller. It could also serve to fill the FileInfo (so user could use the type to feed more than 1 value at once).

I haven't done it because its too many opinions on the reviews and I was afraid of someone complaining. But its the correct way of dealing with such problem (and keeping encapsulation).

Reply:
I am for it.
If you want to avoid changing it now, than at least change the reference to the mutex so it would be a copy and not the original one, and restrict the states access by other method (you can change it to be `nil` if you want in the copy).
However, I think just the copy of the information is better approach.